test/ConnectionTest2.cpp

```diff
@@ -0,0 +1,65 @@
+#include <gtest/gtest.h>
+
+#include "DBConnection2.hpp"
+#include "ConnectionDispatcher.hpp"
+
+namespace {
+
+    int getMySqlConnection2Info(IDBConnection2& dbConnection) {
+        MySqlConnectionDispatcher mySqlConnectionDispatcher;
+        dbConnection.connect(mySqlConnectionDispatcher);
+
+        if (nullptr == mySqlConnectionDispatcher.connection()) {
+            return 0;
+        }
+
+        return mySqlConnectionDispatcher.connection()->query();
+    }
+
+    std::string getMySqlConnection2AdvancedInfo(IDBConnection2& dbConnection) {
+        MySqlConnectionDispatcher mySqlConnectionDispatcher;
+        dbConnection.connect(mySqlConnectionDispatcher);
+
+        if (nullptr == mySqlConnectionDispatcher.connection()) {
+            return std::string();
+        }
+
+        return mySqlConnectionDispatcher.connection()->advancedQuery().serverVersion;
+    }
+
+    int getSqLiteConnection2Info(IDBConnection2& dbconnection) {
+        SqLiteConnectionDispatcher sqLiteConnectionDispatcher;
+        dbconnection.connect(sqLiteConnectionDispatcher);
+
+        if (nullptr == sqLiteConnectionDispatcher.connection()) {
+            return 0;
+        }
+
+        return sqLiteConnectionDispatcher.connection()->query();
+    }
+
+    TEST(MySqlDBConnection2Test, GetInfoTest) {
+        MySqlDBConnection2 connection("4.2.2 MySQL Server", 10);
```

Comment:
@Morheit , two vars names misleading. Name them `mysql_connection` and `sqlite_connection`

Reply:
@malirod , vars renamed